src/uct/api/uct.h

```diff
@@ -1162,9 +1203,57 @@ struct uct_ep_params {
      * 0..(@ref uct_iface_attr_t.dev_num_paths - 1).
      */
     unsigned                            path_index;
+
+    /**
+     * Callback is called when the client side endpoint is bound to network
+     * device and provides related info as defined in
+     * @ref uct_cm_ep_resolve_args_t. This field is only valid if
+     * @ref uct_ep_params_t::sockaddr is set.
+     * @note It is never guaranteed that the callaback will be called. If, for
+     * example, the endpoint goes into error state before issuing the connection
+     * request, the callback will not be invoked.
```

Comment:
This callback is invoked when the remote server address provided in field XYZ is resolved to the local device to be used for connection establishment. 
note
In case of a connection error, this callback will not be invoked; XYZ will be invoked instead
note
This field is mutually exclusive with `@ref uct_ep_params.ABC`